pandas/tests/io/test_excel.py

```diff
@@ -360,6 +360,33 @@ def test_reader_dtype(self, ext):
         with pytest.raises(ValueError):
             actual = self.get_exceldf(basename, ext, dtype={'d': 'int64'})
 
+    def test_reader_dtype_str(self, ext):
+        # GH 20377
+        basename = 'testdtype'
+        actual = self.get_exceldf(basename, ext)
+
+        expected = DataFrame({
+            'a': [1, 2, 3, 4],
+            'b': [2.5, 3.5, 4.5, 5.5],
+            'c': [1, 2, 3, 4],
+            'd': [1.0, 2.0, np.nan, 4.0]}).reindex(
+                columns=['a', 'b', 'c', 'd'])
+
+        tm.assert_frame_equal(actual, expected)
+
+        actual = self.get_exceldf(basename, ext,
+                                  dtype={'a': 'float64',
+                                         'b': 'float32',
+                                         'c': str,
+                                         'd': str})
+
+        expected['a'] = expected['a'].astype('float64')
```

Comment:
move this higher (by the expected), you can simply construct things directly by using e.g. ``Series(...., dtype='float32')`` rather than a list

Reply:
> move this higher (by the expected)

I'm not sure what you mean here.

> you can simply construct things directly by using e.g. Series(...., dtype='float32') rather than a list

First of all, this is copy-paste from the previous test, which was added for https://github.com/pandas-dev/pandas/issues/8212

Do you mean to do
```python
expected = DataFrame({'a': Series([1,2,3,4], dtype='float64'),
                      'b': Series([2.5,3.5,4.5,5.5], dtype='float32'),
                      ...})
```
?
If i use, for example, for the `'c'` column: `Series([1, 2, 3, 4], dtype=str)`, then it will give me `['1', '2', '3', '4']` instead of the expected `['001', '002', '003', '004']`.

Reply:
yes exactly. If you need things like '001', then just do it that way, e.g. ``Series(['001', '002'....])``